react-carousel/src/plugins/infinite.js

```diff
@@ -0,0 +1,129 @@
+import { useRecoilState, useRecoilValue, useSetRecoilState } from 'recoil';
+import { isNil } from 'lodash';
+import times from 'lodash/times';
+import concat from 'lodash/concat';
+import { useEffect } from 'react';
+
+import getChildren from '../tools/getChildren';
+import CAROUSEL_STRATEGIES from '../constants/carouselStrategies';
+import { pluginNames } from '../constants/plugins';
+import {
+  activeSlideIndexState,
+  slideWidthState,
+  slidesState,
+} from '../state/atoms/slideAtoms';
+import {
+  slideMovementState,
+  trackStylesState,
+  trackWidthState,
+} from '../state/atoms/carouselAtoms';
+
+const defaultOptions = {
+  numberOfInfiniteClones: 2,
+};
+
+const infinite = ({ options = defaultOptions, carouselProps }) => {
+  const itemWidth = useRecoilValue(slideWidthState);
+  const children = getChildren(carouselProps.children, carouselProps.slides);
+
+  const getTargetMod = (customValue = null) => {
+    const value = isNil(customValue) ? carouselProps.value : customValue;
```

Comment:
```suggestion
    const value = customValue === null ? carouselProps.value : customValue;
```

Reply:
would it work?

Reply:
Yes, but isNil also checks whether value is undefined